source/ts/app-connection.ts

```diff
@@ -83,7 +83,15 @@ export class AppConnection extends EventEmitter {
     });
 
     if (this.logDirectory && !fs.existsSync(this.logDirectory)) {
-      fs.mkdirSync(this.logDirectory);
+      try {
+        fs.mkdirSync(this.logDirectory, {
+          recursive: true,
+        });
+      } catch (error) {
+        if (error.code !== 'EEXIST') {
+          throw error;
+        }
+      }
```

Comment:
From the [docs](https://nodejs.org/api/fs.html#fsmkdirpath-options-callback), it looks like it won't throw `EEXIST` if `recursive: true`:

> Calling fs.mkdir() when path is a directory that exists results in an error only when recursive is false

So I think we could go possibly go further:

- Remove the `existsSync` check
- Remove the try-catch

```
if (this.logDirectory) {
    fs.mkdirSync(this.logDirectory, {
      recursive: true,
    });
}
```

Reply:
Good shout, done [here](b26026343d791fbd2f0346f031ef1648dee149e7)